controllers/pvc-mutator/storageclass/annotation_test.go

```diff
@@ -0,0 +1,148 @@
+		t.Fatal(err)
+	}
+
+	// Default StorageOS StorageClass.
+	defStosSC := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:  types.UID("storageos-default-uid"),
+			Name: "stos-default",
+		},
+		Provisioner: provisioner.DriverName,
+	}
+
+	// StorageOS StorageClass.
+	stosSC := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:  types.UID("storageos-uid"),
+			Name: "stos",
+		},
+		Provisioner: provisioner.DriverName,
+	}
+
+	// Non-StorageOS StorageClass.
+	notStosSC := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:  types.UID("foo-uid"),
+			Name: "non-stos",
+		},
+		Provisioner: "foo-provisioner",
+	}
+
+	testNamespace := "default"
+
+	testcases := []struct {
+		name         string
+		namespace    string
+		storageClass *storagev1.StorageClass
+		wantErr      bool
+	}{
+		{
+			name:      "empty StorageClass",
+			namespace: testNamespace,
+			wantErr:   true,
+		},
+		{
+			name:         "foreign StorageClass",
+			namespace:    testNamespace,
+			storageClass: notStosSC,
+		},
+		{
+			name:         "default StorageClass",
+			namespace:    testNamespace,
+			storageClass: defStosSC,
+		},
+		{
+			name:         "given StorageClass",
+			namespace:    testNamespace,
+			storageClass: stosSC,
+		},
+	}
+
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			// Create all the above resources and get a k8s client.
+			k8s := fake.NewClientBuilder().WithScheme(scheme).WithObjects(defStosSC, stosSC, notStosSC).Build()
+
+			// Create a AnnotationSetter instance with the fake client.
+			annotationSetter := AnnotationSetter{
+				Client: k8s,
+				log:    ctrl.Log,
+			}
+
+			// Create a pvc
+			pvc := createPVC("pvc1", tc.namespace, tc.storageClass)
+
+			err := annotationSetter.MutatePVC(context.Background(), pvc, tc.namespace)
+			if err != nil {
+				if !tc.wantErr {
+					t.Errorf("got unexpected error: %v", err)
+				}
+				return
+			}
+
+			// Collect annotation to test
+			scAnnotation, ok := pvc.Annotations[AnnotationKey]
+
+			// Validate result
+			switch tc.storageClass.Provisioner {
+			case provisioner.DriverName:
+				if scAnnotation != string(tc.storageClass.UID) {
+					t.Errorf("annotation value got:\n%s\n, want:\n%s", scAnnotation, string(tc.storageClass.UID))
+				}
+			default:
+				if ok {
+					t.Error("annotation found for foreign")
+				}
+			}
+		})
+	}
+}
+
+// createPVC creates and returns a PVC object.
+func createPVC(name, namespace string, storageClass *storagev1.StorageClass) *corev1.PersistentVolumeClaim {
+	pvc := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteOnce,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
```

Comment:
Looks like we are creating similar `createPVC()` in every PVC mutator package.
Maybe separately, in the future, we should have a common test helper package that returns the core PVC instance given name, namespace and such package specific `createPVC()` functions can just wrap their requirements around the core share PVC creation helper function.